cmake/thirdparty/get_spdlog.cmake

```diff
@@ -12,15 +12,39 @@
 # the License.
 # =============================================================================
 
-# Use CPM to find or clone speedlog
+# Use CPM to find or clone speedlog TODO: The logic here should be upstreamed into rapids-logger and
+# probably removed from rapids-cmake. Note that it is not possible to build with two different modes
+# for different projects in the same build environment, so this functionality must be kept
+# independent of the rapids_make_logger function. Alternatively, we could update the cpm calls for
+# spdlog to not promote targets to global scope and instead allow each project to find and configure
+# spdlog independently.
```

Comment:
I'm not sure what we want to do about this. Do we want to allow different projects in the same build to use different spdlog targets? The underlying issue is that rather than having separate dynamic and static targets, spdlog changes the meaning of the spdlog target depending on whether it's compiled dynamically or statically. As a result, you cannot have both within the same CMake build.

Reply:
I don't see why we need both?

Reply:
It's hard to see in this PR. I'm going to punt this to a follow-up. For now I'll stick with using header-only spdlog to get the first piece done.